### PR TITLE
Fix more warnings and enable warnings as errors

### PR DESCRIPTION
**Pull request by FlexW**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -77,9 +77,9 @@ include(GetGitRevisionDescription)
 get_git_head_revision(GIT_REFSPEC GIT_SHA1)
 
 add_definitions(
+    -DQT_DISABLE_DEPRECATED_BEFORE=0
```
Comment (mgallien) [inline CMakeLists.txt:80]: I disagree and according to [QT_DISABLE_DEPRECATED_BEFORE Documentation](https://doc.qt.io/qt-5/qtglobal.html#QT_DISABLE_DEPRECATED_BEFORE) I would have expected something like `QT_DISABLE_DEPRECATED_BEFORE=0x051500`

**Review by mgallien — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (FlexW) [reply CMakeLists.txt:80]: Why do you disagree? If we don't disable the deprecation warnings we can't enable warning as errors. A few deprecation warnings are not solvable at the moment with Qt5. I don't think it helps us in day to day work to see these warnings.

File: cmake/modules/Warnings.cmake [hunk 2]
```diff
@@ -6,7 +6,7 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
 
     # Use this only for Clang
     if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -Wno-long-long -Wno-gnu-zero-variadic-macro-arguments")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall -Wextra -pedantic -Wno-long-long -Wno-gnu-zero-variadic-macro-arguments")
```
[the remark this answers is not in the document]
Comment (FlexW) [reply cmake/modules/Warnings.cmake]: Okay, you have a point with that :). I changed it.

File: CMakeLists.txt (hunk 1 above)
Comment (mgallien) [reply CMakeLists.txt]: is it not possible to for example enable those from Qt 5.14 ?
I guess we really only have a problem with deprecations from QtNetwork that are only solvable with Qt 6
given I may miss something, please correct me if needed

Comment (FlexW) [reply CMakeLists.txt]: I think there are some older ones too. That flag was enabled already for src/gui.

Comment (FlexW) [reply CMakeLists.txt]: and the desktop client does not compile without it.

Comment (mgallien) [reply CMakeLists.txt]: OK something for next Friday !

File: .drone.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ steps:
       path: /drone/build
   commands:
     - cd /drone/build
-    - cmake -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10 -DCMAKE_BUILD_TYPE=Debug -DBUILD_UPDATER=ON -DBUILD_TESTING=1 -DECM_ENABLE_SANITIZERS=address ../src
+    - cmake -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10 -DCMAKE_BUILD_TYPE=Debug -DBUILD_UPDATER=ON -DBUILD_TESTING=1 -DWERROR=ON -DECM_ENABLE_SANITIZERS=address ../src
```
Comment (mgallien) [inline .drone.yml:12]: you can just add `-Werror` by setting it like this
```
cmake -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10 -DCMAKE_BUILD_TYPE=Debug -DBUILD_UPDATER=ON -DBUILD_TESTING=1 -DCMAKE_CXX_FLAGS=-Werror -DECM_ENABLE_SANITIZERS=address ../src
```

File: .drone.yml [hunk 4]
```diff
@@ -53,15 +53,15 @@ steps:
       path: /drone/build
   commands:
     - cd /drone/build
-    - cmake -GNinja -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_C_COMPILER=clang-10 -DCMAKE_CXX_COMPILER=clazy -DCMAKE_BUILD_TYPE=Debug -DBUILD_UPDATER=ON -DBUILD_TESTING=1 -DECM_ENABLE_SANITIZERS=address ../src
+    - cmake -GNinja -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_C_COMPILER=clang-10 -DCMAKE_CXX_COMPILER=clang++-10 -DCMAKE_BUILD_TYPE=Debug -DBUILD_UPDATER=ON -DBUILD_TESTING=1 -DWERROR=ON -DECM_ENABLE_SANITIZERS=address ../src
```
Comment (mgallien) [inline .drone.yml:56]: same comment

File: NEXTCLOUD.cmake [hunk 5]
```diff
@@ -20,6 +20,8 @@ set( WIN_SETUP_BITMAP_PATH  "${CMAKE_SOURCE_DIR}/admin/win/nsi" )
 
 set( MAC_INSTALLER_BACKGROUND_FILE "${CMAKE_SOURCE_DIR}/admin/osx/installer-background.png" CACHE STRING "The MacOSX installer background image")
 
+option(WERROR "Build with warnings as errors" OFF)
```
Comment (mgallien) [inline NEXTCLOUD.cmake:23]: not needed if you see my other comment

**Review by mgallien — CHANGES_REQUESTED** [review]

File: NEXTCLOUD.cmake (hunk 5 above)
Comment (FlexW) [reply NEXTCLOUD.cmake]: done

File: .drone.yml (hunk 4 above)
Comment (FlexW) [reply .drone.yml]: done

File: .drone.yml (hunk 3 above)
Comment (FlexW) [reply .drone.yml]: done

**Review by mgallien — APPROVED** [review]

**Review by allexzander — APPROVED** [review]

**Pull request merged**